packages/core/src/performers/step-performer/StepPerformerPromptCreator.ts

```diff
@@ -249,4 +250,15 @@ export class StepPerformerPromptCreator {
     );
     return steps;
   }
+
+  private previousStepResultOrError(
+    previousStep: PreviousStep,
+    isMostPreviousStep: boolean,
+  ): string {
+    if (previousStep.result && !previousStep.error)
+      return `- Result: ${previousStep.result}`;
+    if (isMostPreviousStep && previousStep.error)
+      return `- Error:\n"${previousStep.error}" \nOccurred in your previous attempt. Try another approach for this step`;
```

Comment:
let's include the error within a block, since it might be a long one
```suggestion
      return `- Error occurred in your previous attempt. Try another approach to perform this step. Error message:\n\`\`\`${previousStep.error}\`\`\``;
```

Reply:
Also let's limit `previousStep.error` to print only the first `2000` chars, then truncate the rest (which is **SUPER** high), just to make sure that it doesn't break our prompt.

Reply:
e.g.
```ts
function truncateString(str, num) {
  if (str.length <= num) {
    return str;
  } else {
    return str.slice(0, num) + "...";
  }
}
```